operate/qa/integration-tests/src/test/java/io/camunda/operate/elasticsearch/FinishedImportingIT.java

```diff
@@ -0,0 +1,219 @@
+    EXPORTER.export(record);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    zeebeImporter.performOneRoundOfImport();
+
+    // when
+    final var record2 = generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(record2);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    // receives 8.7 record and marks partition as finished importing
+    zeebeImporter.performOneRoundOfImport();
+
+    // then
+    // require multiple checks to avoid race condition. If records are written to zeebe indices and
+    // before a refresh, the record reader pulls the import batch is empty so it then says that the
+    // record reader is done when it is not.
+    for (int i = 0;
+        i < ElasticsearchRecordsReader.MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER;
+        i++) {
+      zeebeImporter.performOneRoundOfImport();
+    }
+
+    // the import position for
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .until(() -> isRecordReaderIsCompleted("1-process-instance"));
+  }
+
+  @Test
+  public void shouldMarkMultiplePositionIndexAsCompletedIf870RecordReceived() throws IOException {
+    final var processInstanceRecord = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
+    final var decisionEvalRecord = generateRecord(ValueType.DECISION_EVALUATION, "8.6.0", 1);
+    final var decisionRecord = generateRecord(ValueType.DECISION, "8.6.0", 1);
+    EXPORTER.export(processInstanceRecord);
+    EXPORTER.export(decisionEvalRecord);
+    EXPORTER.export(decisionRecord);
+
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+    zeebeImporter.performOneRoundOfImport();
+
+    // when
+    final var newVersionProcessInstanceRecord =
+        generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(newVersionProcessInstanceRecord);
+    final var newVersionDecisionEvalRecord =
+        generateRecord(ValueType.DECISION_EVALUATION, "8.7.0", 1);
+    EXPORTER.export(newVersionDecisionEvalRecord);
+
+    for (int i = 0;
+        i <= ElasticsearchRecordsReader.MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER;
+        i++) {
+      // simulate existing decision records left to process so it is not marked as completed
+      final var decisionRecord2 = generateRecord(ValueType.DECISION, "8.6.0", 1);
+      EXPORTER.export(decisionRecord2);
+      esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+      zeebeImporter.performOneRoundOfImport();
+    }
+
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .until(
+            () ->
+                isRecordReaderIsCompleted("1-process-instance")
+                    && isRecordReaderIsCompleted("1-decision-evaluation"));
+    Awaitility.await()
+        .during(Duration.ofSeconds(10))
+        .atMost(Duration.ofSeconds(11))
+        .until(() -> !isRecordReaderIsCompleted("1-decision"));
```

Comment:
🔧 Here I'm bit hesitant to do it like this, as we would wait here for 10 seconds right? Could we do it differently? Or maybe just assert it is not completed?

It is always hard to test that something doesn't not happen, so we mostly should avoid it, or verify what we can assert.